CMakeLists.txt

```diff
@@ -104,12 +104,8 @@ set(LICENSE "Mozilla Public License Version 2.0")
 
 # Build, project and include settings
 find_package(Threads REQUIRED)
-if(WITH_DLT_COREDUMPHANDLER OR WITH_DLT_FILETRANSFER)
-    set(ZLIB_LIBRARY "-lz")
-    find_package(ZLIB REQUIRED)
-else()
-    set(ZLIB_LIBRARY "")
-endif()
+find_package(ZLIB REQUIRED)
```

Comment:
can we make this optional? i.e. only require zlib if a cmake option is set to on. Some systems might not have zlib and do not need it either so making this optional would make sure the build is still working there